### PR TITLE
Lets scuttlebots scan for forensics. Gives dets a nerfed version.

### PR DESCRIPTION
**Pull request by Bartimeus973**
## About the PR 
This PR gives syndicate scuttlebots a new ability allowing them to scan for forensics.

scuttlescan

Allows you to turn the detective's detgadget hat into a weaker scuttlebot that doesnt possess the flash ability and has no hands. The detective must now make a choice between wearing the cool detgadget hat, or sending it out in the field. The items stored inside the detgadget hat are restored when you put the goggles back into the scuttlebot.

Goggles are now linked to their own scuttlebot and you can no longer use any goggles on any scuttlebot. You must now use the goggles that came with the bot if you want to turn it off.

Also scuttlebots have a higher pitched scream and can now fart, because sord wanted it and i found it amusing.

## Why's this needed? 
Gives detectives a fun harmless toy to mess with. Scuttlebots are no longer an instant way of figuring out there are spies or that the detective is a traitor.

## Changelog

```changelog
(u)Bartimeus973
(*)Detectives now have a weak scuttlebot of their own! It has taken the shape of the detgadget hat in their locker.
```

### CODE REVIEW

File: code/obj/item/clothing/hats.dm [hunk 1]
```diff
@@ -405,6 +405,15 @@ proc/filter_trait_hats(var/type)
 
 		return ..()
 
+	attack_self (mob/user as mob)
+		user.visible_message("<span class='combat'><b>[user] turns their detgadget hat into a spiffy scuttlebot!</b></span>")
```
Comment (flrsh) [inline code/obj/item/clothing/hats.dm:409]:
```suggestion
		user.visible_message("<span class='combat'><b>[user] turns [his_or_her(user)] detgadget hat into a spiffy scuttlebot!</b></span>")
```

**Review by flrsh — COMMENTED** [review]
Can't think of anything particularly concerning with the buff to be able to pick up normal sized items. @Tarmunora any thoughts?

**Comment by Sovexe** [comment]
> Can't think of anything particularly concerning with the buff to be able to pick up normal sized items. @Tarmunora any thoughts?

You'll be able to carry tasers, batons, activated csabers

**Comment by Tarmunora** [comment]
> > Can't think of anything particularly concerning with the buff to be able to pick up normal sized items. @Tarmunora any thoughts?
> 
> You'll be able to carry tasers, batons, activated csabers, security barriers, etc - and still be able to pass through most doors via squeeze

yeah I'm not feeling this one tbh

**Comment by flrsh** [comment]
> You'll be able to carry tasers, batons, activated csabers, security barriers, etc - and still be able to pass through most doors via squeeze

:sheltersweat: good point

**Comment by Bartimeus973** [comment]
Very well, it seems i'll drop the scuttlebuff and leave the rest of the pr as is.

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**